### DISCUSSION TITLE
是否可以每个页面有单独的meta description？

### DISCUSSION THREAD

**Discussion by WAGFS** [halo-dev/halo#7050] [Q&A]
最近，我的网站遇到了一些SEO问题，显示多个页面存在重复的meta description内容。经过检查，发现主要是authors页面、tags页面及categories页面的描述与主页相同。然而，代码中无法通过`<meta name="description" content="xxxx" />`的方式覆盖默认的description。此外，相关文档似乎也没有提供类似于`spec.description`的变量来进行调整。请问对此类问题有何解决方案？

**Comment by JohnNiang** [comment]
建议提供对应的主题，看看模板文件是否有适配 meta。

**Reply by WAGFS** [reply]
使用的是默认主题https://github.com/halo-dev/theme-earth

**Comment by VoidYun** [comment]
我在使用的时候也遇到了这个问题，目前试过的方法如下：

1. 在需要页面的 `<head>` 中加上 `<meta name="description" th:attr="content=${description}">`
2. 在需要页面的 `<head>` 中加上 `<meta name="description" th:content="${description}">`
3. 使用 js 添加或替换

首先 js 肯定是没问题的，问题就在于 1 和 2

在填写了 系统设置-SEO设置-站点描述 之后，除了 post 和 page 其他页面都会加上这个描述，导致 1 和 2 会出现两个 `<meta name="description">` 的情况，当然 js 也就要从添加变成替换了。

然而使用 js 并不是一个好的实现方式，所以希望有更好的解决方法。

>如果您考虑使用 JavaScript 注入或更改 meta 标记，请谨慎操作。我们建议您尽可能避免使用 JavaScript 注入或更改 meta 标记。如果确有必要这么做，请全面测试您的实现。
>https://developers.google.com/search/docs/crawling-indexing/special-tags?hl=zh-cn#other-points-to-note

我的想法是可以选择性开启和关闭自动添加meta description，一方面不需要现有主题被动兼容，另一方面可以让想要实现的主题主动适配，例如
```html
<meta name="description" th:content="${description ?: site.seo.description}">
```
